Added the page to add songs

WangRemy opened the pull request:
Able to add songs to the Song database model.

WangRemy committed:
Migration for rating model.

Added the date time field so it can be displayed on the
details page. Gave the already added votes my birthdate as
a default holder.

WangRemy committed:
Added Ratings to the detail page.

The ratings for the song are displayed on the details
page ordered by newest  to oldest added.

WangRemy committed:
added html page for add song page

WangRemy committed:
Added add_song function

Added the forms needed to send and save
the title and artist and insert the data into the
Song model in the database.

WangRemy committed:
Added Urls

Added url for the add song page,
also added a url link on all other pages to
redirect back to the home page.

SanderBeekhuis commented on song_rater/views.py:
```diff
@@ -40,3 +40,15 @@ def song_detail(request, id):
     return render(request, 'song_rater/song_detail.html', context)
+
+
+def add_song(request):
+    if request.method == "POST":
+        if request.POST.get('title') and request.POST.get('artist'):
+            add_song = Song()
+            add_song.title = request.POST.get('title')
+            add_song.artist = request.POST.get('artist')
+            add_song.save()
+            return render(request, 'song_rater/add_song.html')
+    else:
+        return render(request, 'song_rater/add_song.html')
```
I think you must be able to Use a GenericView instead of writing this code ourselves.

For example a `CreateView` https://docs.djangoproject.com/en/3.1/ref/class-based-views/

SanderBeekhuis replied:
You did well!

SanderBeekhuis replied:
Nicely done

SanderBeekhuis commented on templates/song_rater/add_song.html:
```diff
@@ -0,0 +1,20 @@
+    <li><a href="{% url 'song_rater:index'%}">Back to homepage</a></li>
+</form>
+</body>
+</html>
```
Trailing newline

SanderBeekhuis commented on templates/song_rater/add_song.html:
```diff
@@ -0,0 +1,20 @@
+   <br/>
+   <br/>
+   <input type="submit" value="Post"/>
+    <li><a href="{% url 'song_rater:index'%}">Back to homepage</a></li>
```
Nice navigation link.

I wouldn't put in inside a `li` tough.  I think it looks weird with only one option.  Up to style tough.

SanderBeekhuis replied:
Thanks

SanderBeekhuis commented on templates/song_rater/song_detail.html:
```diff
@@ -27,5 +27,6 @@ <h2>Rate The Song</h2>
        <td>Sorry, no ratings in this list.</td>
        {% endfor %}
     </ul>
+   <li><a href="{% url 'song_rater:index'%}">Back to homepage</a></li>
```
Nice that you added a link here as well.  Same remark on the `li` element as above

SanderBeekhuis replied:
Also, A "back to songs list" link might make more sense.

SanderBeekhuis commented on templates/song_rater/song_list.html:
```diff
@@ -24,6 +24,7 @@ <h1>List of all Songs</h1>
        <td>Sorry, no songs in this list.</td>
        {% endfor %}
     </table>
+   <li><a href="{% url 'song_rater:index'%}">Back to homepage</a></li>
```
Nice that you added a link here as well. Same remark on the li element as above

SanderBeekhuis replied:
Thumbs up

SanderBeekhuis reviewed (changes requested):
I review only the last 3 commits as these are the ones that differ from #6.  

I like the code. I think my request to use a Generic view is the only major change. 

If i would want to merge this now I couldn't since you continued to work your previous feature branches. It would have been nicer if you started working from `master` again. 

I'm missing tests as well.

WangRemy committed:
add latest ratings header

WangRemy committed:
Cleaned up the link

removed the <li> element

WangRemy committed:
Used GenericView instead of "tedious" code

Removed "tedious" code and used GenericView from
django Generic Views. Changed to correct URL and
updated the html form aswell as the name. Removed add_song.html.

WangRemy committed:
Added testcases for add song page

checks if the link is valid and the page is valid,
then it adds a song via client post and checks if the
title of the song is in the database.

WangRemy committed:
Added testcases for add song page

checks if the link is valid and the page is valid,
then it adds a song via client post and checks if the
title of the song is in the database.

WangRemy committed:
Added confirmation for song add

Added a page to confirm your song has been added to the database,
also added new url and html file.

SanderBeekhuis commented:
Na de merge van #5 heeft deze branch conflicst gekregen.  We moeten hem dus updaten met de nieuwste versie van master.

SanderBeekhuis commented on song_rater/views.py:
```diff
@@ -42,13 +43,11 @@ def song_detail(request, id):
+class AddSong(CreateView):
+    model = Song
+    fields = ['title', 'artist']
+    success_url = "/song_add_succes"
```
The succes URL could also just be the songs list page.  

Save the user a click

SanderBeekhuis replied:
Discussed with @WangRemy  and decided to leave as is.

WangRemy committed:
Merge branch 'master' into add-song

# Conflicts:
#	song_rater/tests.py
#	song_rater/views.py
#	templates/song_rater/song_detail.html

SanderBeekhuis commented on templates/song_rater/song_add_succes.html:
```diff
@@ -0,0 +1,11 @@
+<H1>Your song has been added succesfully!</H1>
+<a href="{% url 'song_rater:index'%}">Back to homepage</a>
+</body>
+</html>
```
Trailing newline :)

WangRemy committed:
Tweak success page

Add url that retuns to the add song page

WangRemy committed:
Used reverse lazy function instead of hard coded url

Makes for a more readable code and easier to
substitute the root url

WangRemy committed:
Added newline to conform to flake8
